### PR TITLE
[Doc] Add index desgin doc

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #6919 
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>

### CODE REVIEW

File: docs/design_docs/index_design.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+the result of the Meta update is obtained from the ETCD, and the `Event.Kv.Version` of the update event is compared
+with the `revision` in the MetaTable. If the `Event.Kv.Version` is greater than the `revision` in the MetaTable, 
+Explain that this update is initiated by IndexNode, and then update the MetaTable in IndexCoord. Since this update
+is initiated by IndexNode, it indicates that this IndexNode has completed this task, so update the load of this 
+IndexNode in NodeManager, and the task amount is reduced by one.
+
+### 8.3.6 assignTaskLoop
+
+`assignTaskLoop` is used to assign index construction tasks. There is a timer here to traverse the MetaTable regularly
+to filter out the tasks that need to be allocated, including unallocated tasks and tasks that have been suspended by
```
Comment (xiaofan-luan) [inline docs/design_docs/index_design.md:105]: including unallocated tasks and tasks that have been failed due to indexNode crash

File: docs/design_docs/index_design.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+original data, the data is deserialized into data blocks, and then cgo is called to build the index. When the index is
+built, the index data is serialized into data blocks, and then written into the file. The directory organization of the
+index file is "indexBuildID" /IndexTaskVersion/partitionID/segmentID/key", where key corresponds to the serialized key
+of index data. After the index is built, record the index file directory in IndexMeta, and then write it to ETCD.
+
+## 8.5 API
+
+### 8.5.1 BuildIndex
+
+Index building is asynchronous, so when an index building request comes, an IndexBuildID is assigned to the task and
```
Comment (xiaofan-luan) [inline docs/design_docs/index_design.md:141]: we need a diagram show the state machine of index building task 
for example:
Start -> process -> failure/success

File: docs/design_docs/index_design.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+### 8.5.1 BuildIndex
+
+Index building is asynchronous, so when an index building request comes, an IndexBuildID is assigned to the task and 
+the task is recorded in Meta. The background process assignTaskLoop will find this task and assign it to IndexNode for
+execution.
+
+### 8.5.2 DropIndex
+
+DropIndex deletes indexes based on IndexID. One IndexID corresponds to the index of an entire column. A column is
```
Comment (xiaofan-luan) [inline docs/design_docs/index_design.md:147]: Are we gonna to support index task cancellation once dropIndex?

File: docs/design_docs/index_design.md (hunk 2 above)
Comment (xiaofan-luan) [reply docs/design_docs/index_design.md:141]: IndexState is the state I mean

File: docs/design_docs/index_design.md (hunk 3 above)
Comment (xiaocai2333) [reply docs/design_docs/index_design.md:154]: Currently, we do not support canceling the index being built, if necessary, we can try it.

**Pull request merged**